Unstable test KcOidcBrokerTest.testAccountManagementLinkIdentity

### Describe the bug

The test failed with the following message when running  tests in https://github.com/keycloak/keycloak/actions/runs/3284788769/jobs/5413106478

```
[ERROR] Tests run: 28, Failures: 0, Errors: 1, Skipped: 1, Time elapsed: 57.874 s <<< FAILURE! - in org.keycloak.testsuite.broker.KcOidcBrokerTest
-- [ERROR] org.keycloak.testsuite.broker.KcOidcBrokerTest.testAccountManagementLinkIdentity  Time elapsed: 9.035 s  <<< ERROR!
-- org.openqa.selenium.NoSuchElementException:
-- Unable to locate element with ID: username
-- For documentation on this error, please visit: http://seleniumhq.org/exceptions/no_such_element.html
-- Build info: version: '3.14.0', revision: 'aacccce0', time: '2018-08-02T20:19:58.91Z'
-- System info: host: 'fv-az284-120', ip: '10.1.0.139', os.name: 'Linux', os.arch: 'amd64', os.version: '5.15.0-1020-azure', java.version: '11.0.16.1'
-- Driver info: driver.version: HtmlUnitDriver
--         at org.jboss.arquillian.graphene.enricher.WebElementUtils$5.getTarget(WebElementUtils.java:127)
--         at org.jboss.arquillian.graphene.proxy.GrapheneProxyHandler.getTarget(GrapheneProxyHandler.java:147)
--         at org.jboss.arquillian.graphene.proxy.GrapheneContextualHandler$1.getTarget(GrapheneContextualHandler.java:218)
--         at org.jboss.arquillian.graphene.proxy.GrapheneContextualHandler$1.invoke(GrapheneContextualHandler.java:169)
--         at org.jboss.arquillian.graphene.intercept.InterceptorBuilder$2.intercept(InterceptorBuilder.java:139)
--         at org.jboss.arquillian.graphene.proxy.InvocationContextImpl.invoke(InvocationContextImpl.java:87)
--         at org.jboss.arquillian.graphene.enricher.SearchContextInterceptor.intercept(SearchContextInterceptor.java:50)
--         at org.jboss.arquillian.graphene.proxy.InvocationContextImpl.invoke(InvocationContextImpl.java:87)
--         at org.jboss.arquillian.graphene.enricher.StaleElementInterceptor$1.apply(StaleElementInterceptor.java:47)
--         at org.jboss.arquillian.graphene.enricher.StaleElementInterceptor$1.apply(StaleElementInterceptor.java:43)
--         at org.openqa.selenium.support.ui.FluentWait.until(FluentWait.java:260)
--         at org.jboss.arquillian.graphene.wait.WebDriverWaitImpl.until(WebDriverWaitImpl.java:109)
--         at org.jboss.arquillian.graphene.enricher.StaleElementInterceptor.intercept(StaleElementInterceptor.java:43)
--         at org.jboss.arquillian.graphene.proxy.InvocationContextImpl.invoke(InvocationContextImpl.java:87)
--         at org.jboss.arquillian.graphene.proxy.GrapheneContextualHandler$2.call(GrapheneContextualHandler.java:241)
--         at org.jboss.arquillian.graphene.context.BrowserActions.performAction(BrowserActions.java:62)
--         at org.jboss.arquillian.graphene.proxy.GrapheneContextualHandler.invoke(GrapheneContextualHandler.java:237)
--         at com.sun.proxy.$Proxy58.clear(Unknown Source)
--         at org.keycloak.testsuite.pages.LoginPage.login(LoginPage.java:88)
--         at jdk.internal.reflect.GeneratedMethodAccessor1185.invoke(Unknown Source)
--         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
--         at java.base/java.lang.reflect.Method.invoke(Method.java:566)
--         at org.jboss.arquillian.graphene.proxy.GrapheneProxyHandler.invokeReal(GrapheneProxyHandler.java:129)
--         at org.jboss.arquillian.graphene.proxy.GrapheneContextualHandler$1.invoke(GrapheneContextualHandler.java:169)
--         at org.jboss.arquillian.graphene.proxy.GrapheneContextualHandler$2.call(GrapheneContextualHandler.java:241)
--         at org.jboss.arquillian.graphene.context.BrowserActions.performAction(BrowserActions.java:62)
--         at org.jboss.arquillian.graphene.proxy.GrapheneContextualHandler.invoke(GrapheneContextualHandler.java:237)
--         at org.jboss.arquillian.graphene.proxy.GrapheneContextualHandler.intercept(GrapheneContextualHandler.java:263)
--         at org.keycloak.testsuite.pages.LoginPage$$EnhancerByGraphene$$4c63faaf.login(<generated>)
--         at org.keycloak.testsuite.broker.AbstractAdvancedBrokerTest.testAccountManagementLinkIdentity(AbstractAdvancedBrokerTest.java:143)
```

### Version

nightly

### Expected behavior

Wait for the login field to appear before logging in

### Actual behavior

It might be that after clicking on the social login link the code hasn't waited long enough for the login field to appear. 

``` 
        // AbstractAdvancedBrokerTest.java
        loginPage.clickSocial(bc.getIDPAlias());
        loginPage.login(bc.getUserLogin(), bc.getUserPassword());
```

### How to Reproduce?

unstable test, not reproducable

--- a/testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/pages/LoginPage.java
+++ b/testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/pages/LoginPage.java
@@ -21,6 +21,7 @@ import org.jboss.arquillian.test.api.ArquillianResource;
 import org.junit.Assert;
 import org.keycloak.testsuite.util.DroneUtils;
 import org.keycloak.testsuite.util.OAuthClient;
+import org.keycloak.testsuite.util.WaitUtils;
 import org.openqa.selenium.By;
 import org.openqa.selenium.NoSuchElementException;
 import org.openqa.selenium.WebElement;
@@ -46,9 +47,6 @@ public class LoginPage extends LanguageComboboxAwarePage {
     @FindBy(id = "input-error")
     private WebElement inputError;
 
-    @FindBy(id = "totp")
-    private WebElement totp;
-
     @FindBy(id = "rememberMe")
     private WebElement rememberMe;
 
@@ -64,14 +62,8 @@ public class LoginPage extends LanguageComboboxAwarePage {
     @FindBy(linkText = "Forgot Password?")
     private WebElement resetPasswordLink;
 
-    @FindBy(linkText = "Username")
-    private WebElement recoverUsernameLink;
-
     @FindBy(className = "alert-error")
     private WebElement loginErrorMessage;
-
-    @FindBy(className = "alert-warning")
-    private WebElement loginWarningMessage;
 
     @FindBy(className = "alert-success")
     private WebElement loginSuccessMessage;
@@ -85,13 +77,24 @@ public class LoginPage extends LanguageComboboxAwarePage {
 
 
     public void login(String username, String password) {
-        usernameInput.clear();
+        clearUsernameInputAndWaitIfNecessary();
         usernameInput.sendKeys(username);
 
         passwordInput.clear();
         passwordInput.sendKeys(password);
 
         clickLink(submitButton);
+    }
+
+    private void clearUsernameInputAndWaitIfNecessary() {
+        try {
+            usernameInput.clear();
+        } catch (NoSuchElementException ex) {
+            // we might have clicked on a social login icon and might need to wait for the login to appear.
+            // avoid waiting by default to avoid the delay.
+            WaitUtils.waitUntilElement(usernameInput).is().present();
+            usernameInput.clear();
+        }
     }
 
     public void login(String password) {
@@ -102,14 +105,14 @@ public class LoginPage extends LanguageComboboxAwarePage {
     }
 
     public void missingPassword(String username) {
-        usernameInput.clear();
+        clearUsernameInputAndWaitIfNecessary();
         usernameInput.sendKeys(username);
         passwordInput.clear();
         clickLink(submitButton);
 
     }
     public void missingUsername() {
-        usernameInput.clear();
+        clearUsernameInputAndWaitIfNecessary();
         clickLink(submitButton);
 
     }
@@ -210,10 +213,6 @@ public class LoginPage extends LanguageComboboxAwarePage {
 
     public void resetPassword() {
         clickLink(resetPasswordLink);
-    }
-
-    public void recoverUsername() {
-        clickLink(recoverUsernameLink);
     }
 
     public void setRememberMe(boolean enable) {

